### PR TITLE
Fix pcg deserialization

### PR DESCRIPTION
**Pull request by tavplubix**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed bug in deserialization of random generator state with might cause some data types such as `AggregateFunction(groupArraySample(N), T))` to behave in a non-deterministic way

### CODE REVIEW

File: tests/queries/0_stateless/01156_pcg_deserialization.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+for engine in "${engines[@]}"
+do
+  $CLICKHOUSE_CLIENT -q "drop table if exists t";
+  $CLICKHOUSE_CLIENT -q "create table t (n UInt8, a1 AggregateFunction(groupArraySample(1), UInt8)) engine=$engine"
+  $CLICKHOUSE_CLIENT -q "insert into t select number % 5 as n, groupArraySampleState(1)(toUInt8(number)) from numbers(10) group by n"
+
+  $CLICKHOUSE_CLIENT -q "select * from t format TSV" | $CLICKHOUSE_CLIENT -q "insert into t format TSV"
+  $CLICKHOUSE_CLIENT -q "select countDistinct(n), countDistinct(a1) from t"
+
```
Comment (tavplubix) [inline tests/queries/0_stateless/01156_pcg_deserialization.sh:17]: It used to produce different results with `Memory` and `MergeTree`/`Log`, I have no ideas why:
https://gist.github.com/tavplubix/8fafec4bca6a1b78087afd99aea85ff9

**Comment by alexey-milovidov** [comment]
@tavplubix See also https://github.com/ClickHouse/ClickHouse/issues/20480

**Comment by alesapin** [comment]
@Mergifyio update

**Comment by tavplubix** [comment]
Fast test - 01156_pcg_deserialization - `Illegal type AggregateFunction(groupArraySample(1), UInt8) of argument for aggregate function uniqExact`
But it worked before merging with master

**Comment by alexey-milovidov** [comment]
@tavplubix It is broken here: https://github.com/ClickHouse/ClickHouse/pull/24523

**Comment by alexey-milovidov** [comment]
@mergifyio update

File: tests/queries/0_stateless/01156_pcg_deserialization.sh (hunk 1 above)
Comment (alexey-milovidov) [reply tests/queries/0_stateless/01156_pcg_deserialization.sh:17]: You did not specify seed, so random seed is selected.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
It should not affect #20480, because quantileDeterministic does not use rng.

**Comment by tavplubix** [comment]
Now the test fails due to another bug:
```
DB::Exception: Conversion from AggregateFunction(groupArraySample, UInt8) to AggregateFunction(groupArraySample(1), UInt8) is not supported
```

```
$ ./clickhouse local -q "select version(), toTypeName(groupArraySampleState(1)(toUInt8(number))) from numbers(10)"
21.7.1.7111	AggregateFunction(groupArraySample(1), UInt8)

$ ./clickhouse local -q "select version(), toTypeName(groupArraySampleState(1)(toUInt8(number))) from numbers(10)"
21.7.1.7128	AggregateFunction(groupArraySample, UInt8)
```

It should not ignore parameters of nested aggregate function. I will try to fix it.

**Pull request merged**